src/picterra/client.py

```diff
@@ -82,22 +82,29 @@ def _wait_until_operation_completes(self, operation_response):
                 raise APIError('Operation %s failed' % operation_id)
             time.sleep(poll_interval)
 
-    def _paginate_through_list(self, resource_endpoint: str):
-        url = self._api_url('%s/?page_number=1' % resource_endpoint)
+    def _paginate_through_list(self, resource_endpoint: str, params: dict = {}):
+        url = self._api_url('%s/' % resource_endpoint)
         data = []
-        while url:
-            logger.debug('Paginating through %s list at page %s' % (resource_endpoint, url))
-            resp = self.sess.get(url)
+        page = 1
+        while True:
+            params['page_number'] = page
+            logger.debug('Paginating through %s list at page %s' % (resource_endpoint, page))
+            resp = self.sess.get(url, params=params)
             if not resp.ok:
                 raise APIError(resp.text)
             r = resp.json()
-            url = r['next']
             count = r['count']
             data += r['results']
+            if not r['next']:
```

Comment:
Is the change from `while url` do `while True` with the `if` fixing a bug of some kind ?

Reply:
If that's not necessary I'd maybe revert, because I kinda feel we should prefer using the `next` url rather than computing it ourselves (and having to keep track of the current page).

Reply:
Checked

Reply:
I added another comment, but I basically wouldn't add bookkeeping (it's always error prone) that we just use us when debugging the library. This is stuff you can add when debugging and remove afterwards. Let's see if one day we want to limit pagination - then we can add this, but for now I'd just keep it simple